data/index.json

```diff
@@ -2,13 +2,15 @@
   "survey_results": [
     {
       "name": "Simple Survey",
+      "surveyId": 1,
       "url": "/survey_results/1.json",
       "participant_count": 6,
       "response_rate": 0.8333333333333334,
       "submitted_response_count": 5
     },
     {
       "name": "Acme Engagement Survey",
+      "surveyId": 2,
```

Comment:
The only reason you put `surveyId` in is to generate a url to the survey. 

So why not just put the Survey url in here? like what's been done with `url`?

Maybe:
```js
"survey_results": [
    {
      "name": "Simple Survey",
      "surveyId": 1,
      "urls": {
          "survey": "/survey/1.json",
          "results": "/survey_results/1.json",
      },
      "participant_count": 6,
      "response_rate": 0.8333333333333334,
      "submitted_response_count": 5
    },
...
```

Have a google for `HATEOAS` - this is a pattern that many companies actually strive for.

This opens you up to questions about backwards compatibility - so if you read up on HATEOAS, and API versioning you'll be set to answer their questions and that'll lead to them being impressed ;)

Just make sure you prepare to get asked about the benefits and drawbacks of this approach